jolt-core/src/jolt/trace/rv.rs

```diff
@@ -493,9 +493,9 @@ impl JoltProvableTrace for RVTraceRow {
     };
 
     // Canonical ordering for memory instructions
-    // 0: rd
-    // 1: rs1
-    // 2: rs2
+    // 0: rs1
```

Comment:
What's the reasoning for this swap? I don't mind just trying to understand.

Reply:
When computing timestamps in `ReadWriteMemory::new`, we process memory ops in this order. If the destination register is equal to one of the source registers, we need to process the write after the read, otherwise the timestamps would be invalid